src/library/dao/playlistdao.cpp

```diff
@@ -431,38 +436,33 @@ void PlaylistDAO::removeTrackFromPlaylist(const int playlistId, const TrackId& t
     }
 
     while (query.next()) {
-    	int position = query.value(query.record().indexOf("position")).toInt();
+        int position = query.value(query.record().indexOf("position")).toInt();
         removeTracksFromPlaylistInner(playlistId, position);
-
     }
-
-    transaction.commit();
-    emit(changed(playlistId));
 }
 
 
-void PlaylistDAO::removeTrackFromPlaylist(const int playlistId, const int position) {
+void PlaylistDAO::removeTrackFromPlaylist(int playlistId, int position) {
     // qDebug() << "PlaylistDAO::removeTrackFromPlaylist"
     //          << QThread::currentThread() << m_database.connectionName();
     ScopedTransaction transaction(m_database);
     removeTracksFromPlaylistInner(playlistId, position);
     transaction.commit();
-    emit(changed(playlistId));
+    emit tracksChanged(QSet<int>{playlistId});
 }
 
-void PlaylistDAO::removeTracksFromPlaylist(const int playlistId, QList<int>& positions) {
+void PlaylistDAO::removeTracksFromPlaylist(int playlistId, QList<int> positions) {
```

Comment:
Because a local, writable copy is need for sorting. First line of the function body.

Reply:
The QList is actually not shared, but moved into this function. The caller never accesses it again after calling the function. The copy happens, because instead of move semantics Qt relies on implicit sharing.

I could add a `std::move` at the caller's site, because QList has a move constructor. This should eliminate the temporary copy.

Reply:
Ah I see, thank you for explaining.

Reply:
I omitted to use `std::move` in the first place because I was recently told not to use it in Qt code ;) What I definitely won't do is checking each and every Qt class before using it. I prefer to use it if it is semantically correct, whether it is supported or not.